src/file-selector.ts

```diff
@@ -121,6 +121,15 @@ function flatten<T>(items: any[]): T[] {
 }
 
 function fromDataTransferItem(item: DataTransferItem) {
+    if (typeof (item as any).getAsFileSystemHandle === 'function') {
+        return (item as any).getAsFileSystemHandle()
+            .then(async (h: any) => {
+                const file = await h.getFile();
+                file.handle = h;
+                return toFileWithPath(file);
+            })
+            .catch((error: {message: any}) => error.message);
```

Comment:
I'd actually return the full error here instead (omit the catch). @tomayac what do you think, does that work for you?

Reply:
Sure, made a suggestion. Is this what you had in mind?

Reply:
Check my other comment.